### ISSUE TITLE
Racetest failure: istio.io/istio/pilot/pkg/proxy/envoy/v2 TestEds

### ISSUE THREAD

**Issue by caohe** [istio/istio#11621]
[https://gubernator.k8s.io/build/istio-circleci/presubmit/racetest/320345](https://gubernator.k8s.io/build/istio-circleci/presubmit/racetest/320345)

```
==================
WARNING: DATA RACE
Read at 0x00c42061e448 by goroutine 104:
  istio.io/istio/pilot/pkg/proxy/envoy/v2.(*DiscoveryServer).periodicRefreshMetrics()
      /go/src/istio.io/istio/pilot/pkg/proxy/envoy/v2/discovery.go:234 +0x15e

Previous write at 0x00c42061e448 by goroutine 292:
  [failed to restore the stack]

Goroutine 104 (running) created at:
  istio.io/istio/pilot/pkg/proxy/envoy/v2.NewDiscoveryServer()
      /go/src/istio.io/istio/pilot/pkg/proxy/envoy/v2/discovery.go:149 +0x27a
  istio.io/istio/pilot/pkg/bootstrap.(*Server).initDiscoveryService()
      /go/src/istio.io/istio/pilot/pkg/bootstrap/server.go:779 +0x640
  istio.io/istio/pilot/pkg/bootstrap.NewServer()
      /go/src/istio.io/istio/pilot/pkg/bootstrap/server.go:227 +0x26f
  istio.io/istio/tests/util.setup()
      /go/src/istio.io/istio/tests/util/pilot_server.go:164 +0x3dd
  istio.io/istio/tests/util.EnsureTestServer()
      /go/src/istio.io/istio/tests/util/pilot_server.go:109 +0x84
  istio.io/istio/pilot/pkg/proxy/envoy/v2_test.initLocalPilotTestEnv()
      /go/src/istio.io/istio/pilot/pkg/proxy/envoy/v2/xds_test.go:140 +0x101
  istio.io/istio/pilot/pkg/proxy/envoy/v2_test.TestAdsReconnectWithNonce()
      /go/src/istio.io/istio/pilot/pkg/proxy/envoy/v2/ads_test.go:29 +0x53
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:777 +0x16d

Goroutine 292 (running) created at:
  istio.io/istio/vendor/google.golang.org/grpc.(*Server).serveStreams.func1()
      /go/src/istio.io/istio/vendor/google.golang.org/grpc/server.go:678 +0xb8
  istio.io/istio/vendor/google.golang.org/grpc/transport.(*http2Server).operateHeaders()
      /go/src/istio.io/istio/vendor/google.golang.org/grpc/transport/http2_server.go:416 +0x1443
  istio.io/istio/vendor/google.golang.org/grpc/transport.(*http2Server).HandleStreams()
      /go/src/istio.io/istio/vendor/google.golang.org/grpc/transport/http2_server.go:456 +0x87a
  istio.io/istio/vendor/google.golang.org/grpc.(*Server).serveStreams()
      /go/src/istio.io/istio/vendor/google.golang.org/grpc/server.go:676 +0x16e
  istio.io/istio/vendor/google.golang.org/grpc.(*Server).handleRawConn.func2()
      /go/src/istio.io/istio/vendor/google.golang.org/grpc/server.go:631 +0x53
  istio.io/istio/vendor/google.golang.org/grpc.(*Server).handleRawConn.func3()
      /go/src/istio.io/istio/vendor/google.golang.org/grpc/server.go:639 +0x34
==================
testing.go:730: race detected during execution of test
```

**Comment by caohe** [comment]
Lint fails on unrelated files

**Comment by howardjohn** [comment]
This was fixed in 1.1/1.2 in https://github.com/istio/istio/pull/10036/files. I don't think we need to worry about a racetest failure in 1.0 anymore
